mobius-rx2/src/test/java/com/spotify/mobius/rx2/TransformersTest.java

```diff
@@ -80,4 +91,78 @@ public void effectPerformerInvokesConsumerOnSchedulerAndPassesTheRequestedEffect
     scheduler.triggerActions();
     assertThat(consumer.getCurrentValue(), is("First Time"));
   }
+
+  @Test
+  public void effectPerformerInvokesFunctionWithReceivedEffectAndEmitsReturnedEvents() {
+    PublishSubject<String> upstream = PublishSubject.create();
+    TestScheduler scheduler = new TestScheduler();
+    Function<String, Integer> function = s -> s.length();
+    TestObserver<Integer> observer =
+        upstream.compose(Transformers.fromFunction(function, scheduler)).test();
+
+    upstream.onNext("Hello");
+    scheduler.triggerActions();
+    observer.assertValue(5);
+  }
+
+  @Test
+  public void effectPerformerInvokesFunctionWithReceivedEffectAndErrorsForUnhandledExceptions() {
+    PublishSubject<String> upstream = PublishSubject.create();
+    TestScheduler scheduler = new TestScheduler();
+    Function<String, Integer> function =
+        s -> {
+          throw new RuntimeException("Something bad happened");
+        };
+    TestObserver<Integer> observer =
+        upstream.compose(Transformers.fromFunction(function, scheduler)).test();
+
+    upstream.onNext("Hello");
+    scheduler.triggerActions();
+    observer.assertError(RuntimeException.class);
+  }
+
+  @Test
+  public void processingLongEffectsDoesNotBlockProcessingShorterEffects() {
```

Comment:
@pettermahlen added a test that makes sure the desired behavior works as intended. If the underlying code is changed to use `map` or anything else instead of `flatMap`, this test will fail.